app/models/event_stream.rb

```diff
@@ -43,6 +43,38 @@ def emit_notifications
     _log.log_backtrace(err)
   end
 
+  def self.group_levels(strings, regexps, event_groups)
```

Comment:
Since this method modifies the parameters, perhaps it should be a `!` method?  Perhaps it just needs a clearer name.  Also, probably should be private.